mysql-test/suite/galera/t/galera_kill_ddl_query.test

```diff
@@ -0,0 +1,611 @@
+#
+# This test checks wbat happens to TOI DDL when the thread is killed.
+# (The KILLs should be ignored if PXC is in the middle of an TOI op or
+# after the TOI has ended).
+#
+
+--source include/have_debug_sync.inc
+--source include/have_innodb.inc
+--source include/galera_cluster.inc
+--source include/count_sessions.inc
+
+--connection node_1
+call mtr.add_suppression("WSREP: TO isolation failed");
+--let $node1_connection_id=`SELECT CONNECTION_ID()`
+
+--connect node_1a, 127.0.0.1, root, , test, $NODE_MYPORT_1
+--let $node1a_connection_id=`SELECT CONNECTION_ID()`
+
+--connect node_1b, 127.0.0.1, root, , test, $NODE_MYPORT_1
+
+#
+# Test 1: Block with FTWRL, then KILL
+# (The command will succeed)
+#
+--echo ################ Test 1 ################
+--connection node_1
+--echo [connection node_1]
+FLUSH TABLES WITH READ LOCK;
+
+# Do the CREATE TABLE, this will block due to the FTWRL
+--connection node_1a
+--echo [connection node_1a]
+--send CREATE TABLE t1 (ID INT PRIMARY KEY);
+
+# Kill the CREATE TABLE query
+--connection node_1b
+--echo [connection node_1b]
+
+# The KILL will fail because we are already in TOI
+--disable_query_log
+--echo # Killing query on node1a
+--error ER_KILL_DENIED_ERROR
```

Comment:
This test case can fail sporadically when there is high load.
```
[  9%] galera.galera_kill_ddl_query             w12 [ fail ]
        Test ended at 2021-02-26 14:40:28

CURRENT_TEST: galera.galera_kill_ddl_query
mysqltest: At line 43: query 'KILL QUERY $node1a_connection_id' succeeded - should have failed with errno 1095...

The result from queries just before the failure was:
call mtr.add_suppression("WSREP: TO isolation failed");
################ Test 1 ################
[connection node_1]
FLUSH TABLES WITH READ LOCK;
[connection node_1a]
CREATE TABLE t1 (ID INT PRIMARY KEY);;
[connection node_1b]
# Killing query on node1a
safe_process[147562]: Child process: 147563, exit: 1
```
Can be reproduced with 
```
./mtr galera_kill_ddl_query{,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,} --mem --parallel=auto
```

Reply:
fixed.